Common/Data/Auxiliary/FactorFile.cs

```diff
@@ -315,7 +315,16 @@ public FactorFile Apply(List<BaseData> data, SecurityExchangeHours exchangeHours
             var lastEntry = SortedFactorFileData.Last().Value;
             factorFileRows.Add(lastEntry);
 
-            var combinedData = GetSplitsAndDividends(data[0].Symbol, exchangeHours).Concat(data)
+            var splitsAndDividends = GetSplitsAndDividends(data[0].Symbol, exchangeHours);
+
+            var combinedData = splitsAndDividends.Concat(data)
+                .DistinctBy(e =>
+                    {
+                        var asSplit = e as Split;
+                        var eventType = asSplit == null ? "Dividend" : "Split";
+                        return $"{e.Symbol.ID}{eventType}{e.Time.ToStringInvariant(DateFormat.EightCharacter)}{e.Value.SmartRounding().Normalize()}";
```

Comment:
Do we need the symbol id in this expression? Isn't everything the same symbol? We basically just want to remove duplicates, so I think something like `${e.GetType().Name}{e.Time.ToStringInvariant(DateFormat.EightCharacter)}"` should suffice